sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/code.py

```diff
@@ -2,9 +2,29 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # ---------------------------------------------------------
 
-from typing import Optional
+from typing import List, Optional, Union
+from pathlib import Path
 
+from azure.ai.ml._utils._pathspec import GitWildMatchPattern
 from ...entities._assets import Code
+from ...entities._component.code import ComponentIgnoreFile
+
+
+class InternalComponentIgnoreFile(ComponentIgnoreFile):
+    _INTERNAL_COMPONENT_CODE_IGNORES = ["*.additional_includes"]
+
+    def __init__(
```

Comment:
as offline discussed, let's refactor base class and extract base_dir.

I think the original implementation is not designed to support "copy content to a temp folder", which is brought by additional includes actually.

Reply:
I have implemented one version for this, may take a review